apps/verticalhome/js/pinned_apps_navigation.js

```diff
@@ -122,14 +134,133 @@
       }
     },
 
+    moveItemUp: function moveItemUp(){
+      var topStyle = parseFloat(this.elemList.style.top) || 0;
+      var itm = this.elements[this.selectedElemIndex];
+      var tmp = this.selectedElemIndex;
+      var parentDiv = itm.parentNode;
+      if(tmp <= 0){
+        return false;
+      }
+      var prevIndex = this.selectedElemIndex-1;
+      var itmPrev = this.elements[prevIndex];
+
+      var selectedId = itm.dataset.index;
+      var prevId = itmPrev.dataset.index;
+
+      app.pinnedAppsManager.items[tmp].index = prevId;
+      app.pinnedAppsManager.items[prevId].index = selectedId;
+
+      var cln = itm.cloneNode(true);
+
+      parentDiv.insertBefore(cln, itmPrev);
+      itm.parentNode.removeChild(itm);
+      this.selectedElemIndex--;
+      this.elemList.style.top =  topStyle + 6 + 'rem';
+      var allItems = document.querySelectorAll('.pinned-app-item');
+      this.elements = Array.prototype.slice.call(allItems);
+    },
+
+    moveItemDown: function moveItemDown(){
+      var topStyle = parseFloat(this.elemList.style.top) || 0;
+      var itm = this.elements[this.selectedElemIndex];
+      var tmp =  this.selectedElemIndex;
+      var parentDiv = itm.parentNode;
+      var nextIndex = this.selectedElemIndex+1;
+
+      if(nextIndex >= this.elements.length-2){
+        return false;
+      }
+      var itmNext = this.elements[nextIndex];
+
+      var selectedId = itm.dataset.index;
+      var nextId = itmNext.dataset.index;
+
+      app.pinnedAppsManager.items[tmp].index = nextId;
+      app.pinnedAppsManager.items[nextId].index = selectedId;
+
+      var cln = itmNext.cloneNode(true);
+
+      parentDiv.insertBefore(cln, itm);
+      itmNext.parentNode.removeChild(itmNext);
+
+      this.selectedElemIndex++;
+      this.elemList.style.top =  topStyle - 6 + 'rem';
+      var allItems = document.querySelectorAll('.pinned-app-item');
+      this.elements = Array.prototype.slice.call(allItems);
+    },
+
+    sortItems: function sortItems(){
+      var items = app.pinnedAppsManager.items;
+      var count = items.length-1;
+
+      for (var i = 0; i < count; i++){
+       for (var j = 0; j < count-i; j++){
+          if (items[j].index> items[j+1].index) {
+            var max = items[j];
+            items[j] = items[j+1];
+            items[j+1] = max;
+          }
+        }
+      }
```

Comment:
You can use there function sort for array with customize callback function. It would be easier for understanding.